module/icp/include/modes/modes.h

```diff
@@ -129,14 +143,22 @@ typedef struct ccm_ctx {
 #define	ccm_copy_to		ccm_common.cc_copy_to
 #define	ccm_flags		ccm_common.cc_flags
 
-#ifdef CAN_USE_GCM_ASM
-typedef enum gcm_impl {
-	GCM_IMPL_GENERIC = 0,
-	GCM_IMPL_AVX,
-	GCM_IMPL_AVX2,
-	GCM_IMPL_MAX,
-} gcm_impl;
-#endif
+#if defined(CAN_USE_GCM_ASM)
+/*
+ * enum gcm_simd_impl holds the types of the implemented gcm asm routines for
+ * the various x86 SIMD extensions. Please note that other parts of the code
+ * depends on the order given below, so do not change the order and append new
+ * implementations at the end, but before GSI_NUM_IMPL.
+ */
+typedef enum gcm_simd_impl {
+	GSI_NONE,
+	GSI_OSSL_AVX,
+	GSI_AVX2,
+	GSI_ISALC_SSE,
+	GSI_NUM_IMPL
```

Comment:
I don't think the name should be changed here, it's not within the scope of this pull. I'd keep all the existing names same unless there's a reason related to isalc support to change it.